ambry-replication/src/main/java/com.github.ambry.replication/ReplicationManager.java

```diff
@@ -566,15 +568,20 @@ private void assignReplicasToThreadPool() {
         String threadIdentity =
             "Replica Thread-" + (dataNodeId.getDatacenterName().equals(datacenter) ? "Intra-" : "Inter") + i
                 + datacenter;
-        ReplicaThread replicaThread =
-            new ReplicaThread(threadIdentity, replicasForThread, factory, clusterMap, correlationIdGenerator,
-                dataNodeId, connectionPool, replicationConfig, replicationMetrics, notification, storeKeyFactory,
-                replicationConfig.replicationValidateMessageStream, metricRegistry, replicatingOverSsl, datacenter,
-                responseHandler);
-        if (replicaThreadPools.containsKey(datacenter)) {
-          replicaThreadPools.get(datacenter).add(replicaThread);
-        } else {
-          replicaThreadPools.put(datacenter, new ArrayList<>(Arrays.asList(replicaThread)));
+        try {
+          ReplicaThread replicaThread =
+              new ReplicaThread(threadIdentity, replicasForThread, factory, clusterMap, correlationIdGenerator,
+                  dataNodeId, connectionPool, replicationConfig, replicationMetrics, notification, storeKeyFactory,
+                  storeKeyConverterFactory.getStoreKeyConverter(),
+                  Utils.getObj(transformerClassName, storeKeyFactory, storeKeyConverterFactory), metricRegistry,
```

Comment:
This looks like transformers in ReplicaThread will use a different storeKeyConverter than the ReplicaThread that has them.  The transformers should be using the same storeKeyConverter (factory should make them for different threads, but the same thread should have just one shared amongst its objects)

Reply:
Also, doesn't ReplicaThread take in a list of transformer?  I think the getObj call here produces just one right?

Reply:
The idea is that the ReplicaThread will batch convert the IDs the transformers will see when it does its metadata stuff, but the transformers will only be able to use that cache from the batch convert if they use the same StoreKeyConverter as ReplicaThread itself

Reply:
Yes - the whole point was to have the Transformer use the same StoreKeyConverter as the passed in StoreKeyConverter. That was my intention when making this change, but I missed that here. I'll fix this right away. This is partly whey we should fix #981.

As for the list of transformers, for now I reverted that change here until we fix #981 (as it seemed odd to pass in a `StoreKeyConverter` and a list of `Transformer`s, since those should be coupled). The `MessageSievingInputStream` takes in a list of `Transformer`s, however that does not take a `StoreKeyConverter` directly.

Reply:
Fixed.